File: 414. Third Maximum Number/Solution.cs
public class Solution {
    public int ThirdMax(int[] nums) {
        
        SortedSet<int> maxNums = new SortedSet<int>();
        
        foreach (int num in nums) { 
            maxNums.Add(num);
            if (maxNums.Count > 3) maxNums.Remove(maxNums.Min);
        }
    
        return (maxNums.Count < 3) ? maxNums.Max : maxNums.Min;
    }
}

File: 414. Third Maximum Number/solution.py
import sortedcontainers

class Solution:
    def thirdMax(self, nums: List[int]) -> int:

        maxNums = sortedcontainers.SortedSet()

        for num in nums: 
            maxNums.add(num)
            if len(maxNums) > 3: 
                maxNums.pop(0)
    
        return maxNums[-1] if (len(maxNums) < 3) else maxNums[0]
